qualtran/serialization/bloq.py

```diff
@@ -49,6 +48,107 @@
 )
 
 
+def _get_sympy_function_type(expr: sympy.Expr) -> int:
+    if isinstance(expr, sympy.core.mul.Mul):
+        return sympy_pb2.Function.MULTIPLICATION
+    if isinstance(expr, sympy.core.add.Add):
+        return sympy_pb2.Function.ADDITION
+    if isinstance(expr, sympy.core.power.Pow):
+        return sympy_pb2.Function.POWER
+    if isinstance(expr, sympy.core.Mod):
+        return sympy_pb2.Function.MOD
+    else:
+        return sympy_pb2.Function.NONE
+
+
+def _get_sympy_from_enum(enum: int):
+    enum_to_sympy = {
+        sympy_pb2.Function.MULTIPLICATION: sympy.core.mul.Mul,
+        sympy_pb2.Function.ADDITION: sympy.core.add.Add,
+        sympy_pb2.Function.POWER: sympy.core.power.Pow,
+        sympy_pb2.Function.MOD: sympy.core.Mod,
+        sympy_pb2.Function.NONE: None,
+    }
+
+    return enum_to_sympy[enum]
+
+
+def _get_sympy_operand(expr: sympy.Expr):
+
+    # Expression is a single variable.
+    if isinstance(expr, sympy.core.symbol.Symbol):
+        return sympy_pb2.Parameter(symbol=str(expr))
+
+    # Expression is an integer
+    if issubclass(expr.__class__, sympy.core.numbers.Integer):
+        result = expr.numerator
+        if not isinstance(result, int):
+            raise NotImplementedError(f"Sympy expression {str(expr)} cannot be serialized.")
+        return sympy_pb2.Parameter(const_int=result)
+
+    # Expression cannot be broken down further, but is a constant.
+    if issubclass(expr.__class__, sympy.core.numbers.Number):
+        if isinstance(expr, sympy.core.numbers.Rational):
+            fraction = sympy_pb2.Fraction(numerator=expr.numerator, denominator=expr.denominator)
+            return sympy_pb2.Parameter(const_rat=fraction)
+
+        # If it is not a rational number, then it must be an irrational constant.
+        return sympy_pb2.Parameter(const_irrat=expr.name)
+
+
+def decompose_sympy(expr: sympy.Expr):
```

Comment:
Can we call this method `sympy_expr_to_proto` ? Also add return type annotations
```suggestion
def sympy_expr_to_proto(expr: sympy.Expr) -> sympy_pb2.Term:
```